### PR TITLE
Parameter Helpers

### PR DESCRIPTION
**Pull request by Jmeyer1292**
The Godel project's working status has grown so far beyond the head of this directory that it's not really feasible or worthwhile to iterate through the commits in the order they were added.

This PR represents the first of a new approach where I will begin adding peripheral components, updates to specific files, and eventually core changes that will bring the repo up to the working status. 

This specific PR adds a set of helper functions for caching parameter messages to the user's filesystem via YAML files with the ability to fall back onto the parameter server. This includes implementations for all four of the major message components.

This PR does not contain the usage of these helpers. That will come in the very next PR.

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
General comment: I think it's a bit weird for these helper specialisations to 'live' in the same package as the message definitions themselves. Message packages don't normally include any additional files.

The helpers are not automatically generated, and are (inherently) only for C++, which is different from the other artefacts in this package (generated, and for all languages supported by ROS). To me it is at least counter-intuitive to look for these in here.

File: param_helpers/include/param_helpers/param_set.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  {
+    YamlFileReader yaml_file (file, ns);
+    ParamInterfaces<Data>::load(yaml_file, data);
+  }
+
+  inline bool checkForYamlFile(const std::string& file)
+  {
+    std::ifstream f (file.c_str());
+    if (f.good()) return true;
+    else return false;
```
Comment (gavanderhoorn) [inline param_helpers/include/param_helpers/param_set.h:37]: I think this is not recommended / allowed in the [ROS C++ Styleguide](http://wiki.ros.org/CppStyleGuide). Perhaps running this package through `ros_lint` would be an idea?

File: param_helpers/package.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    a uniform syntax. 
+  </description>
+
+  <maintainer email="jonathan.meyer@swri.org">Jonathan Meyer</maintainer>
+
+  <license>Apache2</license>
+
+  <buildtool_depend>catkin</buildtool_depend>
+  <build_depend>roscpp</build_depend>
+  <run_depend>roscpp</run_depend>
```
Comment (gavanderhoorn) [inline param_helpers/package.xml:17]: Your `package.xml` also has `yaml-cpp` as a system dependency.

File: param_helpers/package.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  <version>0.1.0</version>
+  <description>
+    Header-only utility library for loading/saving parameter messages to
+    different interfaces such as the ROS Param Server and a YAML file with
+    a uniform syntax. 
+  </description>
+
+  <maintainer email="jonathan.meyer@swri.org">Jonathan Meyer</maintainer>
+
+  <license>Apache2</license>
```
Comment (gavanderhoorn) [inline param_helpers/package.xml:13]: See earlier note about license name.

**Comment by Jmeyer1292** [comment]
Closed as a result of the many recent PRs #64 through #81.

**Pull request closed without merging**